### PR TITLE
Replace kwargs.get with keyword-only arguments

### PR DESCRIPTION
**Pull request by eric-wieser**
This is not an exhaustive replacement, but handles the places where the intent and clarity improvement is obvious

Follows on from #20061

#### Release Notes

- other
  - more function raise `TypeError` when passed invalid keyword-arguments, rather than ignoring them silently

### CODE REVIEW

File: sympy/polys/domains/domain.py [hunk 1]
```diff
@@ -406,15 +406,15 @@ def __getitem__(self, symbols):
...
         from sympy.polys.domains.polynomialring import PolynomialRing
-        return PolynomialRing(self, symbols, kwargs.get("order", lex))
+        return PolynomialRing(self, symbols, order)
 
-    def frac_field(self, *symbols, **kwargs):
+    def frac_field(self, *symbols, order=lex):
         """Returns a fraction field, i.e. `K(X)`. """
         from sympy.polys.domains.fractionfield import FractionField
-        return FractionField(self, symbols, kwargs.get("order", lex))
+        return FractionField(self, symbols, lex)
```
Comment (oscarbenjamin) [inline sympy/polys/domains/domain.py:417]: I guess this should `order` rather than `lex`

**Comment by oscarbenjamin** [comment]
This looks good to me

File: sympy/polys/domains/domain.py (hunk 1 above)
Comment (eric-wieser) [reply sympy/polys/domains/domain.py:417]: Whoops, good catch
```suggestion
        return FractionField(self, symbols, order)
```

**Comment by eric-wieser** [comment]
CI failure is because tests pass a `dims` argument to `polytope_integrate`, which was added in fb7f8814dece71da458c0abe07efc155cbc7c53d and removed in 6c7dac1285454c6c890ae6bf237501fd03550818

**Comment by eric-wieser** [comment]
Alright, CI passes

**Comment by oscarbenjamin** [comment]
Looks good. Thanks

File: sympy/calculus/util.py [hunk 2]
```diff
@@ -622,7 +622,7 @@ def lcim(numbers):
 
     return result
 
-def is_convex(f, *syms, **kwargs):
+def is_convex(f, *syms, domain=S.Reals):
```
Comment (asmeurer) [inline sympy/calculus/util.py:625]: There have been issues in the past where having S objects at the module level forces it to create the instance at import time, which slows down import time at best and makes things hard to debug at worst. Hopefully this doesn't cause too many problems like that. IMO shoving all these things on S was a bad idea.

**Pull request merged**